File: server/manager/admin-js-sdk/form/PayForm.js
'use strict';



/**

*/
class PayForm {

    id:Number;

    gold:Number;
    constructor() {

    }

    formObject({id,gold}):PayForm{
        this.id = id;
        this.gold = gold;
        return this;
    }

    static of({id,gold}):PayForm{
        return new PayForm().formObject({id,gold});
    }

    static form(id:Number,gold:Number):PayForm{
        return new PayForm().formObject({id,gold});
    }
}

export default PayForm;

File: server/project/mj-client/src/test/java/majiang/client/portal/client/admin/form/PayForm.java
package majiang.client.portal.client.admin.form;

import org.forkjoin.apikit.core.*;

import java.util.*;
import java.util.Map.Entry;
import java.util.AbstractMap.SimpleImmutableEntry;

/**

 */
public class PayForm implements ApiMessage {

	private int id;

	private int gold;

	public PayForm() {
	}

	public PayForm(int id, int gold) {
		this.id = id;
		this.gold = gold;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getGold() {
		return gold;
	}

	public void setGold(int gold) {
		this.gold = gold;
	}

	@Override
	public List<Entry<String, Object>> encode(String parent, List<Entry<String, Object>> $list) {

		$list.add(new SimpleImmutableEntry<String, Object>(parent + "id", id));

		$list.add(new SimpleImmutableEntry<String, Object>(parent + "gold", gold));

		return $list;
	}

	@Override
	public String toString() {
		return "PayForm [id=" + id + ",gold=" + gold + ", ]";
	}
}